src/v/cluster/partition_balancer_planner.cc

```diff
@@ -134,12 +143,13 @@ class partition_balancer_planner::request_context {
     uint64_t _planned_moves_size_bytes = 0;
     size_t _failed_actions_count = 0;
     absl::node_hash_set<model::ntp> _cancellations;
+    ss::abort_source& _as;
 };
 
-void partition_balancer_planner::init_per_node_state(
+ss::future<> partition_balancer_planner::init_per_node_state(
```

Comment:
nit: this is usually very quick, so yields are hardly needed

Reply:
hmm thats fair, undone.